protobuf-lite/src/main/java/io/grpc/protobuf/lite/ProtoLiteUtils.java

```diff
@@ -78,6 +78,18 @@ public static void setExtensionRegistry(ExtensionRegistryLite newRegistry) {
     globalRegistry = checkNotNull(newRegistry, "newRegistry");
   }
 
+  /**
+   * Local cache of buffers to use for parsing.  ThreadLocal used a WeakReference internally, so
```

Comment:
I was wrong in an important way. It doesn't use WeakRef for thread id. Instead, it uses it for the ThreadLocal, which is moot. It stores the map as a field on the Thread and it is set to `null` on Thread completion, so these values could be GC'd as soon as the Thread exits.

If under memory pressure though, these values will not be GC'd, because there are strong references to them within a `Map<WeakRef<ThreadLocal>, Object>` (`ThreadLocalMap`). The only lazy cleaning is for when `ThreadLocal`s themselves are GC'd, which we don't care about.